docs/interfaces.md

```diff
@@ -0,0 +1,315 @@
+# Interfaces
+
+Here you find a list of interfaces we support and won't change without further notice.
+
+
+## Models
+
+We can't guarantee stability for whole models. However certain fields are considered stable.
+
+For more information about a model and it's fields, please consult the source code.
+
+### General
+
+All models share a common set of fields:
+
+* `created_at`
+* `modified_at`
+* `created_by_user`
+* `created_by_group`
+* `history` --> manager for accessing historical records
+
+
+### caluma_form.models.Form
+
+#### Fields
+
+* `slug` / `pk`
+* `name`
+* `description`
+* `meta`
+* `is_published`
+* `is_archived`
+* `questions`
+* `source`
+
+### caluma_form.models.FormQuestion
+
+#### Fields
+
+* `id` / `pk`
+* `form`
+* `question`
+* `sort`
+
+#### Methods
+
+* `natural_key()`
+
+
+### caluma_form.models.Question
+
+#### Fields
+
+* `slug` / `pk`
+* `label`
+* `type`
+* `is_required`
+* `is_hidden`
+* `is_archived`
+* `placeholder`
+* `info_text`
+* `static_content`
+* `configuration`
+* `meta`
+* `data_source`
+* `options`
+* `row_form`
+* `sub_form`
+* `source`
+* `format_validators`
+
+#### Properties
+
+* `min_length`
+* `max_length`
+* `max_value`
+* `min_value`
+
+#### Methods
+
+* `empty_values()`
```

Comment:
I'd remove this - it's only used by the validation internally and I don't assume it has a good reason to be used by extensions